doc/features/promise-callback/README.md

```diff
@@ -0,0 +1,40 @@
+# Promise and callback-based API
+
+The driver supports both [promises][promise] and callbacks for the asynchronous methods exposed in the `Client` and
+`Metadata` prototypes, you can choose the approach that suits your needs.
+
+## Promise-based API
+
+```javascript
+client.execute('SELECT name, email FROM users')
+  .then(result => console.log('User with email %s', result.rows[0].email));
+```
+
+When a `callback` is not provided as the last argument, the driver will return a `Promise`, without the need to 
+_promisify_ the driver module. Returned promises are instances of [`Promise` global object][promise] and are created
+using the default constructor: `new Promise(executor)`.
+
+In case you want the driver use a third party `Promise` module (ie: [bluebird][bluebird]) to create the promise
+instances, you can optionally provide your own factory method when creating the `Client` instance, for example:
+
+```javascript
+const BbPromise = require('bluebird');
+const client = new Client({
+  contactPoints: contactPoints,
+  promiseFactory: BbPromise.fromCallback
```

Comment:
Tried this out and it didn't seem to work since what ultimately seems to happen is promiseFactory is passed a `function executor(resolve, reject)` and [`Promise.fromCallback`](http://bluebirdjs.com/docs/api/promise.fromcallback.html) expects a function callback that takes an error first:

> The resolver function is passed a callback that expects to be called back according to error-first node conventions.

Using `Promise.prototype.apply` seemed to work for me (although there's probably a better way).